components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -2913,6 +2916,175 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
         }
     }
 
+    async createOrgAuthProvider(
+        ctx: TraceContext,
+        { entry }: GitpodServer.CreateOrgAuthProviderParams,
+    ): Promise<AuthProviderEntry> {
+        traceAPIParams(ctx, {}); // entry contains PII
+
+        // map params to a new provider
+        const newProvider = <AuthProviderEntry.NewEntry>{
+            host: entry.host,
+            type: entry.type,
+            clientId: entry.clientId,
+            clientSecret: entry.clientSecret,
+            ownerId: entry.ownerId,
+            organizationId: entry.organizationId,
+        };
+
+        let user = this.checkAndBlockUser("createOrgAuthProvider");
+        let team = await this.getTeam(ctx, newProvider.organizationId || "");
+        if (!team) {
+            throw new ResponseError(ErrorCodes.BAD_REQUEST, "Invalid organizationId");
+        }
+
+        await this.guardWithFeatureFlag("orgGitAuthProviders", team);
+
+        // Since this is a create, ensure they're not creating it as someone else
+        if (user.id !== newProvider.ownerId) {
```

Comment:
Had a question here of if we could drop `ownerId` from the params for this endpoint, and set it to the authenticated user's id instead of requiring it to be the authenticated user? Same goes for the update call.

Reply:
Let's remove it from params. Sounds ok. 

Internally, we can either continue using the id of auth'd user, or we can store a marker value to avoid confusion in code paths of personal Git Auth providers. Something like `ownerId: "owned-by-org"` should work.